vertical-pod-autoscaler/pkg/utils/vpa/capping.go

```diff
@@ -300,3 +304,71 @@ func getBoundaryRecommendation(recommendation apiv1.ResourceList, container apiv
 		apiv1.ResourceMemory: *memMaxRequest,
 	}
 }
+
+func applyPodLimitRange(resources []vpa_types.RecommendedContainerResources,
+	pod *apiv1.Pod, limitRange apiv1.LimitRangeItem, resourceName apiv1.ResourceName,
+	get func(vpa_types.RecommendedContainerResources) *apiv1.ResourceList) []vpa_types.RecommendedContainerResources {
+	minLimit := limitRange.Min[resourceName]
+	maxLimit := limitRange.Max[resourceName]
+	defaultLimit := limitRange.Default[resourceName]
+
+	var sumLimit resource.Quantity
+	for i, item := range pod.Spec.Containers {
+		if i >= len(resources) {
+			continue
+		}
+		limit := item.Resources.Limits[resourceName]
+		request := item.Resources.Requests[resourceName]
+		recommendation := (*get(resources[i]))[resourceName]
+		containerLimit, _ := getProportionalResourceLimit(resourceName, &limit, &request, &recommendation, &defaultLimit)
+		if containerLimit != nil {
+			sumLimit.Add(*containerLimit)
+		}
+	}
+	if minLimit.Cmp(sumLimit) <= 0 && (maxLimit.IsZero() || maxLimit.Cmp(sumLimit) >= 0) {
+		return resources
+	}
+	var targetTotalLimit resource.Quantity
+	if minLimit.Cmp(sumLimit) > 0 {
+		targetTotalLimit = minLimit
+	}
+	if !maxLimit.IsZero() && maxLimit.Cmp(sumLimit) < 0 {
+		targetTotalLimit = maxLimit
+	}
+	for i := range pod.Spec.Containers {
+		limit := (*get(resources[i]))[resourceName]
+		cappedContainerRequest, _ := scaleQuantityProportionally(&limit, &sumLimit, &targetTotalLimit)
+		(*get(resources[i]))[resourceName] = *cappedContainerRequest
+	}
+	return resources
+}
+
+func (c *cappingRecommendationProcessor) capProportionallyToPodLimitRange(
+	containerRecommendations []vpa_types.RecommendedContainerResources, pod *apiv1.Pod) ([]vpa_types.RecommendedContainerResources, error) {
+	podLimitRange, err := c.limitsRangeCalculator.GetPodLimitRangeItem(pod.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("error obtaining limit range: %s", err)
+	}
+	if podLimitRange == nil {
+		return containerRecommendations, nil
+	}
+	getLower := func(r vpa_types.RecommendedContainerResources) *apiv1.ResourceList {
```

Comment:
I've mentioned in the PR before that it is actually not needed to compute lower and upper as well, I was mistaken.

Reply:
Done.